src/File.php

```diff
@@ -891,8 +891,12 @@ public function getAbsoluteURL()
      * @param bool $grant Ensures that the url for any protected assets is granted for the current user.
      * @return string
      */
-    public function getURL($grant = true)
+    public function getURL($grant = false)
     {
+        if ($this->File->canViewFile()) {
+            $grant = true;
+        }
```

Comment:
Is there any reason why we'd use `$this->File->canViewFile()` over regular `$this->canView()`?

We should also perform a `!$grant` check first to prevent unnecessary logic

Reply:
DONE